python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2645,3 +2647,31 @@ def get_latest_data_version_record(
             materialization = next(iter(materializations), None)
 
         return materialization or observation
+
+    @public
+    def get_latest_materialization_code_versions(
+        self, asset_keys: Iterable[AssetKey]
+    ) -> Mapping[AssetKey, Optional[str]]:
+        """Returns the code version used for the latest materialization of each of the provided
+        assets.
+
+        Args:
+            asset_keys (Iterable[AssetKey]): The asset keys to find latest materialization code
+                versions for.
+
+        Returns:
+            Mapping[AssetKey, Optional[str]]: A dictionary with a key for each of the provided asset
+                keys. The values will be None if the asset has no materializations. If an asset does
+                not have a code version explicitly assigned to its definitions but was
+                materialization, Dagster assigns a randomly-generated string as its code version.
+        """
+        result: Dict[AssetKey, Optional[str]] = {}
+        latest_materialization_events = self.get_latest_materialization_events(asset_keys)
+        for asset_key in asset_keys:
+            event = latest_materialization_events.get(asset_key)
+            if event is None:
+                result[asset_key] = None
+            else:
+                result[asset_key] = (event.tags or {}).get(CODE_VERSION_TAG)
```

Comment:
Rather than directly accessing the `CODE_VERSION_TAG`, I suggest using `get_data_provenance_from_entry(event).code_version`. More future-proof against changes to code version persistence.

Reply:
Got it - made this change